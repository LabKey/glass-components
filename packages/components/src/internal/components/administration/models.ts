interface Member {
    id: number;
    name: string;
    type: string;
}

interface Group {
    groupName: string;
    members: Member[];
}

export interface GroupMembership {
    [key: string]: Group;
}
