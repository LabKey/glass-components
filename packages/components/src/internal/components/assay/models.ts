/*
 * Copyright (c) 2019 LabKey Corporation
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { immerable, produce } from 'immer';

import { Operation } from '../../../public/QueryColumn';

import { AssayDefinitionModel } from '../../AssayDefinitionModel';
import { LoadingState } from '../../../public/LoadingState';

import { SelectInputChange } from '../forms/input/SelectInput';

import { AssayWizardModel } from './AssayWizardModel';

export interface AssayPropertiesPanelProps {
    operation: Operation;
    model: AssayWizardModel;
    onChange: (fieldValues: any, isChanged?: boolean) => void;
    onWorkflowTaskChange?: SelectInputChange;
}

export class AssayUploadResultModel {
    [immerable] = true;

    readonly assayId: number;
    readonly batchId: number;
    readonly runId: number;
    readonly success: boolean;
    readonly successurl?: string;
    readonly jobId?: number;

    constructor(values?: Partial<AssayUploadResultModel>) {
        Object.assign(this, values);
    }
}

export class AssayStateModel {
    [immerable] = true;

    readonly definitions: AssayDefinitionModel[];
    readonly definitionsError: string;
    readonly definitionsLoadingState: LoadingState;
    readonly protocolError: string;
    readonly protocolLoadingState: LoadingState;

    constructor(values?: Partial<AssayStateModel>) {
        Object.assign(this, values);

        this.definitions = this.definitions ?? [];
        this.definitionsLoadingState = this.definitionsLoadingState ?? LoadingState.INITIALIZED;
        this.protocolLoadingState = this.protocolLoadingState ?? LoadingState.INITIALIZED;
    }

    getById(assayRowId: number): AssayDefinitionModel {
        return this.definitions.find(def => def.id === assayRowId);
    }

    getByName(assayName: string): AssayDefinitionModel {
        const lowerName = assayName.toLowerCase();
        return this.definitions.find(def => def.name.toLowerCase() === lowerName);
    }

    getDefinitionsByTypes(included?: string[], excluded?: string[]): AssayDefinitionModel[] {
        if (!included && !excluded) return this.definitions;

        const lowerIncluded = included?.join('|').toLowerCase().split('|');
        const lowerExcluded = excluded?.join('|').toLowerCase().split('|');

        return this.definitions.filter(def => {
            let include = true;
            if (included?.length > 0) {
                include = lowerIncluded.indexOf(def.type.toLowerCase()) !== -1;
            }
            if (excluded?.length > 0) {
                include = lowerExcluded.indexOf(def.type.toLowerCase()) === -1;
            }
            return include;
        });
    }

    mutate(props: Partial<AssayStateModel>): AssayStateModel {
        return produce<AssayStateModel>(this, draft => {
            Object.assign(draft, props);
        });
    }
}
