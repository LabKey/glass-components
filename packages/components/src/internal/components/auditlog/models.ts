/*
 * Copyright (c) 2016-2018 LabKey Corporation. All rights reserved. No portion of this work may be reproduced in
 * any form or by any electronic or mechanical means without written permission from LabKey Corporation.
 */
import { fromJS, Map, Record, List } from 'immutable';

import { ASSAYS_KEY, SAMPLES_KEY } from '../../app/constants';

export class AuditDetailsModel extends Record({
    rowId: undefined,
    comment: undefined,
    eventUserId: undefined,
    eventDateFormatted: undefined,
    oldData: undefined,
    newData: undefined,
    userComment: undefined,
}) {
    declare rowId?: number;
    declare comment?: string;
    declare eventUserId?: number;
    declare eventDateFormatted?: string;
    declare oldData?: Map<string, string>;
    declare newData?: Map<string, string>;
    declare userComment?: string;

    static create(raw: any): AuditDetailsModel {
        return new AuditDetailsModel({
            ...raw,
            oldData: raw.oldData ? fromJS(raw.oldData) : undefined,
            newData: raw.newData ? fromJS(raw.newData) : undefined,
        });
    }

    getActionLabel(): string {
        if (this.isUpdate()) return 'Updated';
        else if (this.isInsert()) return 'Created';
        else if (this.isDelete()) return 'Deleted';
        else return 'Updated';
    }

    isUpdate(): boolean {
        return this.oldData && this.newData && this.oldData.size > 0 && this.newData.size > 0;
    }

    isInsert(): boolean {
        return this.oldData && this.newData && this.oldData.size === 0 && this.newData.size > 0;
    }

    isDelete(): boolean {
        return this.oldData && this.newData && this.oldData.size > 0 && this.newData.size === 0;
    }
}

export class TimelineEventModel extends Record({
    rowId: undefined,
    eventType: undefined,
    subEventType: undefined,
    summary: undefined,
    user: undefined,
    eventUserId: undefined,
    timestamp: undefined,
    eventTimestamp: undefined,
    entity: undefined,
    entitySeparator: ' - ',
    metadata: undefined,
    oldData: undefined,
    newData: undefined,
    userComment: undefined,
}) {
    declare rowId?: number;
    declare eventType?: string; // sample, assay, workflow, inventory, etc
    declare subEventType?: string; // for example, inventory item vs inventory storage location
    declare summary?: string;
    declare user?: Map<string, any>;
    declare eventUserId?: number;
    declare timestamp?: Map<string, any>;
    declare eventTimestamp?: any;
    declare entity?: Map<string, any>;
    declare entitySeparator?: string;
    declare metadata?: List<Map<string, any>>;
    declare oldData?: Map<string, string>;
    declare newData?: Map<string, string>;
    declare userComment?: string;

    constructor(values?: { [key: string]: any }) {
        super(values);
    }

    getRowKey() {
        return this.eventType + '|' + this.rowId;
    }

    // timezoneStr used for jest test only, to accommodate teamcity timezone difference
    static create(raw: any, timezoneStr?: string): TimelineEventModel {
        const fields = {} as TimelineEventModel;
        fields.rowId = raw['rowId'];
        fields.eventType = raw['eventType'];
        fields.subEventType = raw['subEventType'];
        fields.summary = raw['summary'];
        fields.user = fromJS(raw['user']);
        fields.eventUserId = raw['user']['value'];
        fields.timestamp = fromJS(raw['timestamp']);
        if (raw['timestamp']['value'])
            fields.eventTimestamp = timezoneStr
                ? new Date(raw['timestamp']['value'] + ' ' + timezoneStr)
                : new Date(raw['timestamp']['value']);
        fields.entity = fromJS(raw['entity']);
        fields.entitySeparator = raw['entitySeparator'];
        fields.userComment = raw['userComment'];

        if (raw.metadata) {
            const metaRows = [];
            for (const [key, value] of Object.entries(raw.metadata)) {
                metaRows.push({ field: key, value: fromJS(value) });
            }
            fields.metadata = fromJS(metaRows);
        }

        if (raw.oldData) fields.oldData = fromJS(raw.oldData);
        if (raw.newData) fields.newData = fromJS(raw.newData);

        return new TimelineEventModel(fields);
    }

    getAuditDetailsModel(): AuditDetailsModel {
        if (!this.oldData && !this.newData) return undefined;

        return new AuditDetailsModel({
            rowId: this.rowId,
            oldData: this.oldData,
            newData: this.newData,
            userComment: this.userComment,
        });
    }

    isSameEntity(event: TimelineEventModel): boolean {
        return this.entity && event.entity && this.entity.get('value') === event.entity.get('value');
    }

    getComment(): string {
        if (!this.metadata) return undefined;
        const commentField = this.metadata.find(metadataRow => metadataRow.get('field').toLowerCase() === 'comment');
        if (commentField) return commentField.get('value');
        return undefined;
    }

    getIcon(): string {
        let icon = this.eventType;

        if (ASSAYS_KEY === this.eventType) icon = 'assay';
        else if (this.eventType === 'inventory') {
            const summary = this.summary.toLowerCase();
            if (summary.indexOf('added to') > -1 || summary.indexOf('added location') > -1) icon = 'storage_insert';
            else if (summary.indexOf('discarded') > -1 || summary.indexOf('location deleted') > -1)
                icon = 'storage_remove';
            else if (summary.indexOf('checked in') > -1) icon = 'storage_checkin';
            else if (summary.indexOf('checked out') > -1) icon = 'storage_checkout';
            else if (summary.indexOf('moved') > -1) icon = 'storage_move';
            // for storage update events not associated with check in, use samples icon
            else icon = SAMPLES_KEY;
        }

        return icon;
    }
}

export interface TimelineGroupedEventInfo {
    firstEvent: TimelineEventModel;
    isCompleted: boolean;
    lastEvent: TimelineEventModel;
}
