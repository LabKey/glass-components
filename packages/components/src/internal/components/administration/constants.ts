import { PermissionRoles } from '@labkey/api';
import { Map } from 'immutable';

const ROLE_ACTION_DESCRIPTION =
    'and creating and editing sample types, source types, assay designs, and job templates.';

export const SECURITY_ROLE_DESCRIPTIONS = Map<string, string>([
    [
        PermissionRoles.ApplicationAdmin,
        'Application Administrators have full control over the application. This includes user management, permission assignments, ' +
            ROLE_ACTION_DESCRIPTION,
    ],
    [
        PermissionRoles.ProjectAdmin,
        'Project Administrators have full control over the application in this project container. This includes user management, permission assignments, ' +
            ROLE_ACTION_DESCRIPTION,
    ],
    [
        PermissionRoles.FolderAdmin,
        'Folder Administrators have control over the folder specific pieces of the application. This includes permission assignments, ' +
            ROLE_ACTION_DESCRIPTION,
    ],
    [
        PermissionRoles.Editor,
        'Editors may add new information or edit data related to samples, sources, assays, jobs, but not storage locations.',
    ],
    [PermissionRoles.Reader, 'Readers have a read-only view of the application.'],
]);

export const HOSTED_APPLICATION_SECURITY_ROLES = Map<string, string>([
    [PermissionRoles.Editor, 'Editor'],
    [PermissionRoles.EditorWithoutDelete, 'Editor without Delete'],
    [PermissionRoles.Reader, 'Reader'],
]);

export const SITE_SECURITY_ROLES = Map<string, string>([
    [PermissionRoles.ApplicationAdmin, 'Application Administrator'],
]);

export const APPLICATION_SECURITY_ROLES = Map<string, string>([
    [PermissionRoles.ProjectAdmin, 'Project Administrator'],
    [PermissionRoles.FolderAdmin, 'Folder Administrator'],
    [PermissionRoles.Editor, 'Editor'],
    [PermissionRoles.EditorWithoutDelete, 'Editor without Delete'],
    [PermissionRoles.Reader, 'Reader'],
]);


