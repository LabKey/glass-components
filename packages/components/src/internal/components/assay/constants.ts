import { SCHEMAS } from '../../schemas';

export const GENERAL_ASSAY_PROVIDER_NAME = 'General';

export const PLATE_METADATA_COLUMN = 'PlateMetadata';
export const PLATE_TEMPLATE_COLUMN = 'PlateTemplate';

// Columns are required for us to render the WorkflowTask in EditableDetails components
export const WORKFLOW_TASK_PROPERTIES_REQUIRED_COLUMNS = ['WorkflowTask', 'WorkflowTask/Run', 'Protocol/RowId'];

export const RUN_PROPERTIES_REQUIRED_COLUMNS = SCHEMAS.CBMB.concat(
    'Name',
    'RowId',
    'ReplacesRun',
    'ReplacedByRun',
    'DataOutputs',
    'DataOutputs/DataFileUrl',
    'Batch',
    'Folder'
)
    .concat(WORKFLOW_TASK_PROPERTIES_REQUIRED_COLUMNS)
    .toList();
