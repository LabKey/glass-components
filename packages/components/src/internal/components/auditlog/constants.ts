import { Query } from '@labkey/api';

export type AuditQuery = {
    containerFilter?: Query.ContainerFilter;
    hasDetail?: boolean;
    label: string;
    value: string;
};

export const ATTACHMENT_AUDIT_QUERY: AuditQuery = { value: 'attachmentauditevent', label: 'Attachment Events' };
export const DOMAIN_AUDIT_QUERY: AuditQuery = { value: 'domainauditevent', label: 'Domain Events' };
export const DOMAIN_PROPERTY_AUDIT_QUERY: AuditQuery = {
    value: 'domainpropertyauditevent',
    label: 'Domain Property Events',
};
export const DATA_UPDATE_AUDIT_QUERY: AuditQuery = {
    value: 'queryupdateauditevent',
    label: 'Data Update Events',
    hasDetail: true,
};
export const INVENTORY_AUDIT_QUERY: AuditQuery = {
    value: 'inventoryauditevent',
    label: 'Storage Management Events',
    hasDetail: true,
};
export const LIST_AUDIT_QUERY: AuditQuery = { value: 'listauditevent', label: 'List Events' };
export const GROUP_AUDIT_QUERY: AuditQuery = {
    value: 'groupauditevent',
    label: 'Roles and Assignment Events',
    containerFilter: Query.ContainerFilter.allFolders,
};
export const PROJECT_AUDIT_QUERY: AuditQuery = { value: 'containerauditevent', label: 'Project Events' };
export const SAMPLE_TYPE_AUDIT_QUERY: AuditQuery = { value: 'samplesetauditevent', label: 'Sample Type Events' };
export const SAMPLE_TIMELINE_AUDIT_QUERY: AuditQuery = {
    value: 'sampletimelineevent',
    label: 'Sample Timeline Events',
    hasDetail: true,
};
export const USER_AUDIT_QUERY: AuditQuery = {
    value: 'userauditevent',
    label: 'User Events',
    containerFilter: Query.ContainerFilter.allFolders,
};
export const ASSAY_AUDIT_QUERY: AuditQuery = { value: 'experimentauditevent', label: 'Assay Events' };
export const WORKFLOW_AUDIT_QUERY: AuditQuery = {
    value: 'samplesworkflowauditevent',
    label: 'Sample Workflow Events',
    hasDetail: true,
};
export const SOURCE_AUDIT_QUERY: AuditQuery = { value: 'sourcesauditevent', label: 'Sources Events', hasDetail: true };
export const COMMON_AUDIT_QUERIES: AuditQuery[] = [
    ATTACHMENT_AUDIT_QUERY,
    DOMAIN_AUDIT_QUERY,
    DOMAIN_PROPERTY_AUDIT_QUERY,
    DATA_UPDATE_AUDIT_QUERY,
    INVENTORY_AUDIT_QUERY,
    LIST_AUDIT_QUERY,
    GROUP_AUDIT_QUERY,
    SAMPLE_TYPE_AUDIT_QUERY,
    SAMPLE_TIMELINE_AUDIT_QUERY,
    USER_AUDIT_QUERY,
];

export const NOTEBOOK_AUDIT_QUERY: AuditQuery = {
    value: 'LabBookEvent',
    label: 'Notebook Events',
};

export const NOTEBOOK_REVIEW_AUDIT_QUERY: AuditQuery = {
    value: 'NotebookEvent',
    label: 'Notebook Review Events',
};

export const REGISTRY_AUDIT_QUERY: AuditQuery = { value: 'RegistryEvent', label: 'Registry Events' };

export const AUDIT_EVENT_TYPE_PARAM = 'eventType';
