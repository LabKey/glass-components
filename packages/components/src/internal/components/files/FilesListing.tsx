import React, { PureComponent, ReactNode } from 'react';

import { List, Set } from 'immutable';

import { IFile } from './models';

export interface FilesListingProps {
    canDelete?: boolean;
    files: List<IFile>;
    getFilePropertiesEditTrigger?: (file: IFile) => ReactNode;
    headerText?: string;
    noFilesMessage: string;
    onDelete?: (fileName: string) => void;
    onFileSelection: (event) => void;
    selectedFiles: Set<string>;
    useFilePropertiesEditTrigger?: boolean;
}

interface State {
    confirmDeletionSet: Set<string>;
}

export class FilesListing extends PureComponent<FilesListingProps, State> {
    constructor(props: FilesListingProps) {
        super(props);
        this.state = { confirmDeletionSet: Set<string>() };
    }

    toggleDeleteConfirmation = (fileName: string): void => {
        const { confirmDeletionSet } = this.state;

        this.setState({
            confirmDeletionSet: confirmDeletionSet.has(fileName)
                ? confirmDeletionSet.delete(fileName)
                : confirmDeletionSet.add(fileName),
        });
    };

    deleteFile = (fileName: string): void => {
        this.setState(state => ({ confirmDeletionSet: state.confirmDeletionSet.delete(fileName) }));
        this.props.onDelete?.(fileName);
    };

    render() {
        const {
            files,
            headerText,
            noFilesMessage,
            useFilePropertiesEditTrigger,
            getFilePropertiesEditTrigger,
            canDelete,
            onFileSelection,
            selectedFiles,
        } = this.props;
        const { confirmDeletionSet } = this.state;

        if (!files || files.size === 0) return <div>{noFilesMessage}</div>;

        return (
            <div>
                {headerText && <div className="file-listing--header bottom-spacing">{headerText}</div>}
                {files.map((file, key) => {
                    const { description, downloadUrl, name } = file;
                    const confirmDelete = confirmDeletionSet.has(name);

                    return (
                        <div className="component file-listing-row--container" key={key}>
                            <div className="detail-component">
                                <div className="row">
                                    <div className="col-xs-4 file-listing-icon--container">
                                        <input
                                            checked={selectedFiles.contains(name)}
                                            name={name}
                                            onClick={onFileSelection}
                                            type="checkbox"
                                        />
                                        <i className={file.iconFontCls + ' file-listing-icon'} />
                                    </div>
                                    <div className="col-xs-3 file-listing-info">
                                        <strong>
                                            <a href={downloadUrl} title={name}>
                                                <div className="file-listing-filename">{name}</div>
                                            </a>
                                        </strong>
                                        <br />
                                        <span className="file-listing-info-light">
                                            <span className="test-loc-file-created-by">{file.createdBy}</span>
                                            <span className="test-loc-file-created">{file.created}</span>
                                        </span>
                                    </div>

                                    <div className="col-xs-5">{description}</div>

                                    <span className="pull-right">
                                        {useFilePropertiesEditTrigger && getFilePropertiesEditTrigger(file)}
                                        {canDelete && confirmDelete ? (
                                            <span>
                                                <div className="pull-right">Permanently delete?</div>
                                                <br />
                                                <span className="file-listing-canceldel-icon">
                                                    <button
                                                        className="btn btn-default"
                                                        onClick={() => this.toggleDeleteConfirmation(name)}
                                                        type="button"
                                                    >
                                                        Cancel
                                                    </button>
                                                </span>
                                                <button
                                                    className="btn btn-danger"
                                                    onClick={() => this.deleteFile(name)}
                                                    type="button"
                                                >
                                                    Delete File
                                                </button>
                                            </span>
                                        ) : (
                                            canDelete && (
                                                <span
                                                    className="pull-right file-listing-delete"
                                                    onClick={() => this.toggleDeleteConfirmation(name)}
                                                >
                                                    <i className="fa fa-times-circle file-listing-delete-icon" />
                                                </span>
                                            )
                                        )}
                                    </span>
                                </div>
                            </div>
                        </div>
                    );
                }).toArray()}
            </div>
        );
    }
}
