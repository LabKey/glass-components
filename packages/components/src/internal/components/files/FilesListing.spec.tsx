import React from 'react';
import { List, Set } from 'immutable';
import { shallow } from 'enzyme';

import { FILES_DATA } from '../../../test/data/constants';

import { IFile } from './models';
import { FilesListing } from './FilesListing';

describe('<FilesListing>', () => {
    test('no files', () => {
        const wrapper = shallow(
            <FilesListing
                noFilesMessage="No files for you"
                onFileSelection={jest.fn()}
                selectedFiles={Set<string>()}
                files={List<IFile>()}
            />
        );
        expect(wrapper.text()).toBe('No files for you');
        expect(wrapper).toMatchSnapshot();
    });
    test('with files custom header', () => {
        const wrapper = shallow(
            <FilesListing
                headerText="Custom header"
                noFilesMessage="No files for you"
                onFileSelection={jest.fn()}
                selectedFiles={Set<string>()}
                files={FILES_DATA}
            />
        );
        expect(wrapper.find('div.file-listing--header')).toHaveLength(1);
        expect(wrapper.find('div.file-listing--header').text()).toBe('Custom header');
        expect(wrapper).toMatchSnapshot();
    });
    test('with files not deletable', () => {
        const wrapper = shallow(
            <FilesListing
                noFilesMessage="No files for you"
                onFileSelection={jest.fn()}
                selectedFiles={Set<string>()}
                files={FILES_DATA}
            />
        );
        expect(wrapper.find('div.file-listing-row--container')).toHaveLength(FILES_DATA.size);
        expect(wrapper).toMatchSnapshot();
    });
    test('with files deletable', () => {
        const wrapper = shallow(
            <FilesListing
                noFilesMessage="No files for you"
                onFileSelection={jest.fn()}
                canDelete={true}
                onDelete={jest.fn()}
                selectedFiles={Set<string>()}
                files={FILES_DATA}
            />
        );
        expect(wrapper.find('span.file-listing-delete')).toHaveLength(FILES_DATA.size);
        expect(wrapper).toMatchSnapshot();
    });
});
