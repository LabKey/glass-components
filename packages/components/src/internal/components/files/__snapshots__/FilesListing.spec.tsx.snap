// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilesListing> no files 1`] = `
<div>
  No files for you
</div>
`;

exports[`<FilesListing> with files custom header 1`] = `
<div>
  <div
    className="file-listing--header bottom-spacing"
  >
    Custom header
  </div>
  <div
    className="component file-listing-row--container"
    key="0"
  >
    <div
      className="detail-component"
    >
      <div
        className="row"
      >
        <div
          className="col-xs-4 file-listing-icon--container"
        >
          <input
            checked={false}
            name="exam.xlsx"
            onClick={[MockFunction]}
            type="checkbox"
          />
          <i
            className="fa fa-file-excel-o file-listing-icon"
          />
        </div>
        <div
          className="col-xs-3 file-listing-info"
        >
          <strong>
            <a
              href="/labkey/sm/sampleManagement-downloadAttachments.view?jobId=282&names=exam.xlsx&returnUrl=%2Flabkey%2Fgpat3%2FsampleManagement-appDev.view%23%2Fworkflow%2F282%2Ffiles"
              title="exam.xlsx"
            >
              <div
                className="file-listing-filename"
              >
                exam.xlsx
              </div>
            </a>
          </strong>
          <br />
          <span
            className="file-listing-info-light"
          >
            <span
              className="test-loc-file-created-by"
            >
              Vader
            </span>
            <span
              className="test-loc-file-created"
            >
              2019-11-08 12:26:30.064
            </span>
          </span>
        </div>
        <div
          className="col-xs-5"
        >
          i'm an excel file
        </div>
        <span
          className="pull-right"
        />
      </div>
    </div>
  </div>
  <div
    className="component file-listing-row--container"
    key="1"
  >
    <div
      className="detail-component"
    >
      <div
        className="row"
      >
        <div
          className="col-xs-4 file-listing-icon--container"
        >
          <input
            checked={false}
            name="xray.gif"
            onClick={[MockFunction]}
            type="checkbox"
          />
          <i
            className="fa fa-file-image-o file-listing-icon"
          />
        </div>
        <div
          className="col-xs-3 file-listing-info"
        >
          <strong>
            <a
              href="/labkey/sm/sampleManagement-downloadAttachments.view?jobId=282&names=xray.gif&returnUrl=%2Flabkey%2Fgpat3%2FsampleManagement-appDev.view%23%2Fworkflow%2F282%2Ffiles"
              title="xray.gif"
            >
              <div
                className="file-listing-filename"
              >
                xray.gif
              </div>
            </a>
          </strong>
          <br />
          <span
            className="file-listing-info-light"
          >
            <span
              className="test-loc-file-created-by"
            >
              Skywalker
            </span>
            <span
              className="test-loc-file-created"
            >
              2019-11-14 15:47:51.931
            </span>
          </span>
        </div>
        <div
          className="col-xs-5"
        >
          i'm gif
        </div>
        <span
          className="pull-right"
        />
      </div>
    </div>
  </div>
  <div
    className="component file-listing-row--container"
    key="2"
  >
    <div
      className="detail-component"
    >
      <div
        className="row"
      >
        <div
          className="col-xs-4 file-listing-icon--container"
        >
          <input
            checked={false}
            name="report.json"
            onClick={[MockFunction]}
            type="checkbox"
          />
          <i
            className="fa fa-file-o file-listing-icon"
          />
        </div>
        <div
          className="col-xs-3 file-listing-info"
        >
          <strong>
            <a
              href="/labkey/sm/sampleManagement-downloadAttachments.view?jobId=282&names=sreport.json&returnUrl=%2Flabkey%2Fgpat3%2FsampleManagement-appDev.view%23%2Fworkflow%2F282%2Ffiles"
              title="report.json"
            >
              <div
                className="file-listing-filename"
              >
                report.json
              </div>
            </a>
          </strong>
          <br />
          <span
            className="file-listing-info-light"
          >
            <span
              className="test-loc-file-created-by"
            >
              Vader
            </span>
            <span
              className="test-loc-file-created"
            >
              2019-11-14 15:48:11.472
            </span>
          </span>
        </div>
        <div
          className="col-xs-5"
        >
          i'm a json file
        </div>
        <span
          className="pull-right"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<FilesListing> with files deletable 1`] = `
<div>
  <div
    className="component file-listing-row--container"
    key="0"
  >
    <div
      className="detail-component"
    >
      <div
        className="row"
      >
        <div
          className="col-xs-4 file-listing-icon--container"
        >
          <input
            checked={false}
            name="exam.xlsx"
            onClick={[MockFunction]}
            type="checkbox"
          />
          <i
            className="fa fa-file-excel-o file-listing-icon"
          />
        </div>
        <div
          className="col-xs-3 file-listing-info"
        >
          <strong>
            <a
              href="/labkey/sm/sampleManagement-downloadAttachments.view?jobId=282&names=exam.xlsx&returnUrl=%2Flabkey%2Fgpat3%2FsampleManagement-appDev.view%23%2Fworkflow%2F282%2Ffiles"
              title="exam.xlsx"
            >
              <div
                className="file-listing-filename"
              >
                exam.xlsx
              </div>
            </a>
          </strong>
          <br />
          <span
            className="file-listing-info-light"
          >
            <span
              className="test-loc-file-created-by"
            >
              Vader
            </span>
            <span
              className="test-loc-file-created"
            >
              2019-11-08 12:26:30.064
            </span>
          </span>
        </div>
        <div
          className="col-xs-5"
        >
          i'm an excel file
        </div>
        <span
          className="pull-right"
        >
          <span
            className="pull-right file-listing-delete"
            onClick={[Function]}
          >
            <i
              className="fa fa-times-circle file-listing-delete-icon"
            />
          </span>
        </span>
      </div>
    </div>
  </div>
  <div
    className="component file-listing-row--container"
    key="1"
  >
    <div
      className="detail-component"
    >
      <div
        className="row"
      >
        <div
          className="col-xs-4 file-listing-icon--container"
        >
          <input
            checked={false}
            name="xray.gif"
            onClick={[MockFunction]}
            type="checkbox"
          />
          <i
            className="fa fa-file-image-o file-listing-icon"
          />
        </div>
        <div
          className="col-xs-3 file-listing-info"
        >
          <strong>
            <a
              href="/labkey/sm/sampleManagement-downloadAttachments.view?jobId=282&names=xray.gif&returnUrl=%2Flabkey%2Fgpat3%2FsampleManagement-appDev.view%23%2Fworkflow%2F282%2Ffiles"
              title="xray.gif"
            >
              <div
                className="file-listing-filename"
              >
                xray.gif
              </div>
            </a>
          </strong>
          <br />
          <span
            className="file-listing-info-light"
          >
            <span
              className="test-loc-file-created-by"
            >
              Skywalker
            </span>
            <span
              className="test-loc-file-created"
            >
              2019-11-14 15:47:51.931
            </span>
          </span>
        </div>
        <div
          className="col-xs-5"
        >
          i'm gif
        </div>
        <span
          className="pull-right"
        >
          <span
            className="pull-right file-listing-delete"
            onClick={[Function]}
          >
            <i
              className="fa fa-times-circle file-listing-delete-icon"
            />
          </span>
        </span>
      </div>
    </div>
  </div>
  <div
    className="component file-listing-row--container"
    key="2"
  >
    <div
      className="detail-component"
    >
      <div
        className="row"
      >
        <div
          className="col-xs-4 file-listing-icon--container"
        >
          <input
            checked={false}
            name="report.json"
            onClick={[MockFunction]}
            type="checkbox"
          />
          <i
            className="fa fa-file-o file-listing-icon"
          />
        </div>
        <div
          className="col-xs-3 file-listing-info"
        >
          <strong>
            <a
              href="/labkey/sm/sampleManagement-downloadAttachments.view?jobId=282&names=sreport.json&returnUrl=%2Flabkey%2Fgpat3%2FsampleManagement-appDev.view%23%2Fworkflow%2F282%2Ffiles"
              title="report.json"
            >
              <div
                className="file-listing-filename"
              >
                report.json
              </div>
            </a>
          </strong>
          <br />
          <span
            className="file-listing-info-light"
          >
            <span
              className="test-loc-file-created-by"
            >
              Vader
            </span>
            <span
              className="test-loc-file-created"
            >
              2019-11-14 15:48:11.472
            </span>
          </span>
        </div>
        <div
          className="col-xs-5"
        >
          i'm a json file
        </div>
        <span
          className="pull-right"
        >
          <span
            className="pull-right file-listing-delete"
            onClick={[Function]}
          >
            <i
              className="fa fa-times-circle file-listing-delete-icon"
            />
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<FilesListing> with files not deletable 1`] = `
<div>
  <div
    className="component file-listing-row--container"
    key="0"
  >
    <div
      className="detail-component"
    >
      <div
        className="row"
      >
        <div
          className="col-xs-4 file-listing-icon--container"
        >
          <input
            checked={false}
            name="exam.xlsx"
            onClick={[MockFunction]}
            type="checkbox"
          />
          <i
            className="fa fa-file-excel-o file-listing-icon"
          />
        </div>
        <div
          className="col-xs-3 file-listing-info"
        >
          <strong>
            <a
              href="/labkey/sm/sampleManagement-downloadAttachments.view?jobId=282&names=exam.xlsx&returnUrl=%2Flabkey%2Fgpat3%2FsampleManagement-appDev.view%23%2Fworkflow%2F282%2Ffiles"
              title="exam.xlsx"
            >
              <div
                className="file-listing-filename"
              >
                exam.xlsx
              </div>
            </a>
          </strong>
          <br />
          <span
            className="file-listing-info-light"
          >
            <span
              className="test-loc-file-created-by"
            >
              Vader
            </span>
            <span
              className="test-loc-file-created"
            >
              2019-11-08 12:26:30.064
            </span>
          </span>
        </div>
        <div
          className="col-xs-5"
        >
          i'm an excel file
        </div>
        <span
          className="pull-right"
        />
      </div>
    </div>
  </div>
  <div
    className="component file-listing-row--container"
    key="1"
  >
    <div
      className="detail-component"
    >
      <div
        className="row"
      >
        <div
          className="col-xs-4 file-listing-icon--container"
        >
          <input
            checked={false}
            name="xray.gif"
            onClick={[MockFunction]}
            type="checkbox"
          />
          <i
            className="fa fa-file-image-o file-listing-icon"
          />
        </div>
        <div
          className="col-xs-3 file-listing-info"
        >
          <strong>
            <a
              href="/labkey/sm/sampleManagement-downloadAttachments.view?jobId=282&names=xray.gif&returnUrl=%2Flabkey%2Fgpat3%2FsampleManagement-appDev.view%23%2Fworkflow%2F282%2Ffiles"
              title="xray.gif"
            >
              <div
                className="file-listing-filename"
              >
                xray.gif
              </div>
            </a>
          </strong>
          <br />
          <span
            className="file-listing-info-light"
          >
            <span
              className="test-loc-file-created-by"
            >
              Skywalker
            </span>
            <span
              className="test-loc-file-created"
            >
              2019-11-14 15:47:51.931
            </span>
          </span>
        </div>
        <div
          className="col-xs-5"
        >
          i'm gif
        </div>
        <span
          className="pull-right"
        />
      </div>
    </div>
  </div>
  <div
    className="component file-listing-row--container"
    key="2"
  >
    <div
      className="detail-component"
    >
      <div
        className="row"
      >
        <div
          className="col-xs-4 file-listing-icon--container"
        >
          <input
            checked={false}
            name="report.json"
            onClick={[MockFunction]}
            type="checkbox"
          />
          <i
            className="fa fa-file-o file-listing-icon"
          />
        </div>
        <div
          className="col-xs-3 file-listing-info"
        >
          <strong>
            <a
              href="/labkey/sm/sampleManagement-downloadAttachments.view?jobId=282&names=sreport.json&returnUrl=%2Flabkey%2Fgpat3%2FsampleManagement-appDev.view%23%2Fworkflow%2F282%2Ffiles"
              title="report.json"
            >
              <div
                className="file-listing-filename"
              >
                report.json
              </div>
            </a>
          </strong>
          <br />
          <span
            className="file-listing-info-light"
          >
            <span
              className="test-loc-file-created-by"
            >
              Vader
            </span>
            <span
              className="test-loc-file-created"
            >
              2019-11-14 15:48:11.472
            </span>
          </span>
        </div>
        <div
          className="col-xs-5"
        >
          i'm a json file
        </div>
        <span
          className="pull-right"
        />
      </div>
    </div>
  </div>
</div>
`;
